gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -1216,14 +1303,53 @@ var gpii = fluid.registerNamespace("gpii");
                     gpiiKey: undefined,
                     actionResults: {},
                     originalSettings: {},
-                    appliedSolutions: {}
+                    currentSettings: {},
+                    currentPreferences: {}
                 }
             });
             transaction.commit();
         });
         return restorePromise;
     };
 
+     /**
+      * Read setting values of one or more solutions.
+      * @param {Object} that - An instance of gpii.lifecycleManager.
+      * @param {Object} solutions - Solutions with setting to be read from the system.
+      * @return {Promise} A promise whose resolved value is the result of setting values.
+      */
+    gpii.lifecycleManager.processRead = function (that, solutions) {
+        var handlerGetPromises = [];
+        fluid.each(solutions, function (solution, solutionID) {
+            var newSolution = gpii.lifecycleManager.transformSolutionSettings(solution, gpii.settingsHandlers.changesToSettings);
+            fluid.each(newSolution.settingsHandlers, function (handlerSpec) {
+                // If supportedSettings is not specified, don't attempt to get settings
+                if (!handlerSpec.supportedSettings) {
+                    return;
+                }
+                handlerSpec.settings = handlerSpec.supportedSettings;
+
+                // TODO: The code below is to work around an issue with "gpii.windows.spiSettingsHandler" settings
```

Comment:
Can we add a JIRA reference for what the issue is, which hopefully has an explanation of how we plan to fix the settings handler

Reply:
@sgithens, is there a JIRA describing what the issue is? This workaround is copied from [your snapshotter implementation](https://github.com/fluid-project/sjrk-story-telling/pull/47). Thanks.